src/xb-builder-node.c

```diff
@@ -1107,3 +1151,43 @@ xb_builder_node_export (XbBuilderNode *self, XbNodeExportFlags flags, GError **e
 		return NULL;
 	return g_string_free (g_steal_pointer (&xml), FALSE);
 }
+
+/**
+ * xb_builder_node_add_token:
+ * @self: a #XbBuilderNode
+ * @token: a new token
+ *
+ * Adds a token to the builder node.
+ *
+ * Since: 0.3.1
+ **/
+void
+xb_builder_node_add_token (XbBuilderNode *self, const gchar *token)
+{
+	XbBuilderNodePrivate *priv = GET_PRIVATE (self);
+
+	g_return_if_fail (self != NULL);
+	g_return_if_fail (token != NULL);
+
+	if (priv->tokens == NULL)
+		priv->tokens = g_ptr_array_new_with_free_func (g_free);
+	g_ptr_array_add (priv->tokens, g_strdup (token));
+}
+
+/**
+ * xb_builder_node_get_tokens:
+ * @self: a #XbBuilderNode
+ *
+ * Gets the tokens of the builder node.
```

Comment:
Definitely needs some docs to explain what ‘tokens’ are. I suggest linking to the top-level documentation section on it.

Reply:
Added I think